ggshield/core/config/config.py

```diff
@@ -40,6 +55,7 @@ def __init__(self, config_path: Optional[Path] = None):
         self.user_config, self._config_path = UserConfig.load(config_path=config_path)
         self.auth_config = AuthConfig.load()
         self._cmdline_instance_name = None
+        self.dotenv_vars: Set[str] = set()
```

Comment:
Can you rename `dotenv_vars` to `_dotenv_vars` and declare its type with its `_cmdline_instance_name` and `_config_path` friends at the top of the class?

Also adding a comment explaining what this variable is for would be useful.